diagnosis/clusterhealthcheck.sh

```diff
@@ -0,0 +1,230 @@
+# Check if etcd is running on all nodes
+
+log_level -i "Checking for etcd running on all nodes in the cluster"
+for IP in $HOST_LIST
+do
+    HOST_NAME=$(ssh -q -t $USER_NAME@$IP 'echo "$(hostname)"')
+    log_level -i "Checking for etcd running on [$HOST_NAME]"
+    ETCD_STATUS=$(ssh -q -t $USER_NAME@$IP "sudo service etcd status | grep 'Active'")
+    
+    if [[ $ETCD_STATUS == *"active"* ]]; then
+        log_level -i "ETCD is active on [$HOST_NAME]. ETCD test passed"
+    else
+        log_level -e "ETCD is not active on [$HOST_NAME]. ETCD test passed"
+    fi
+done
+
+# ##############################################################
+# Check if kubectl is installed and configured correctly on master nodes
+
+log_level -i "Checking if Kubectl is installed and configured on master nodes"
+for IP in $HOST_LIST
+do
+    HOST_NAME=$(ssh -q -t $USER_NAME@$IP 'echo "$(hostname)"')
+    log_level -i "$HOST_NAME"
+    
+    if [[ $HOST_NAME == *"master"* ]]; then
+        log_level -i "Checking for kubectl on [$HOST_NAME]"
+        KUBECTL_COMMAND=$(ssh -q -t $USER_NAME@$IP "kubectl get nodes")
+        if [[ $KUBECTL_COMMAND == *"NAME"* ]]; then
+            log_level -i "Kubectl is installed and configued on master node. Kubectl test passed"
+        else
+            log_level -e "Kubectl is not installed and configued on master node. Kubectl test failed"
+        fi
+    else
+        log_level -i "Host [$HOST_NAME] is not master, skipping test"
+    fi
+done
+
+# ##############################################################
+# Check if kube-system containers are running in docker on nodes
+
+log_level -i "Checking for kubernetes containers running on nodes"
+MASTER_CONTAINERS="kube-scheduler kube-controller kube-apiserver kube-addon-manager"
+NODE_CONTAINERS="azure-ip-masq-agent kube-flannel kube-proxy"
+for IP in $HOST_LIST
+do
+    HOST_NAME=$(ssh -q -t $USER_NAME@$IP 'echo "$(hostname)"')
+    CONTAINER_LIST=$(ssh -q -t $USER_NAME@$IP 'LIST=$(docker container ls --format="{{.Names }}" --filter status="running"); echo $LIST')
+    
+    if [[ $HOST_NAME == *"master"* ]]; then
+        log_level -i "Checking for master containers"
+        for CONTAINER in $MASTER_CONTAINERS
+        do
+            if [[ $CONTAINER_LIST == *"$CONTAINER"* ]]; then
+                log_level -i "$CONTAINER running on [$HOST_NAME]. Test passed"
+            else
+                log_level -e "$CONTAINER not running on [$HOST_NAME]. Test failed"
+            fi
+        done
+        log_level -i "Checking for node containers"
+        for CONTAINER in $NODE_CONTAINERS
+        do
+            if [[ $CONTAINER_LIST == *"$CONTAINER"* ]]; then
+                log_level -i "$CONTAINER running on [$HOST_NAME]. Test passed"
+            else
+                log_level -e "$CONTAINER not running on [$HOST_NAME]. Test failed"
+            fi
+        done
+    else
+        log_level -i "Checking for node containers"
```

Comment:
no need to duplicate this block of code

Reply:
both  agent node containers and master node containers run on the master node